hydro_deploy/core/src/custom_service.rs

```diff
@@ -125,17 +126,20 @@ impl HydroflowSource for CustomClientPort {
         panic!("Custom services cannot be used as the server")
     }
 
-    fn record_server_config(&mut self, config: ServerConfig) {
-        self.client_port = Some(config);
+    fn record_server_config(&self, config: ServerConfig) {
+        self.client_port
+            .set(config)
+            .map_err(drop) // `ServerConfig` doesn't implement `Debug` for `.expect()`.
+            .expect("Cannot call `record_server_config()` multiple times.");
```

Comment:
Hopefully this is true? `instantiate_reverse(` also calls this, so only one of the two can be called as well

Reply:
Yep, checks out.